misc/thread_tools.c

```diff
@@ -245,7 +245,14 @@ int mp_cancel_get_fd(struct mp_cancel *c)
 {
     pthread_mutex_lock(&c->lock);
     if (c->wakeup_pipe[0] < 0) {
+#if defined(__GNUC__) && !defined(__clang__)
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Wstringop-overflow="
+#endif
         mp_make_wakeup_pipe(c->wakeup_pipe);
```

Comment:
nit: This is quite ugly, but I don't have any better suggestions.

Maybe something like `(int *)&c->wakeup_pipe` could remove the warning? Hard to say since the warning is completely bogus so I have no clue what triggered it.

Reply:
I can try indeed.

Reply:
Doesn't change anything https://github.com/mpv-player/mpv/actions/runs/6519372807/job/17705781867?pr=12420